chatsky/llm/filters.py

```diff
@@ -0,0 +1,56 @@
+"""
+Filters.
+---------
+This module contains a collection of basic functions for history filtering to avoid cluttering LLMs context window.
+"""
+
+from chatsky.core.message import Message
+from chatsky.core.context import Context
+from pydantic import BaseModel
+import abc
+
+
+class BaseFilter(BaseModel, abc.ABC):
+    """
+    Base class for all message history filters.
+    """
+
+    @abc.abstractmethod
+    def __call__(self, ctx: Context, request: Message, response: Message, model_name: str) -> bool:
+        """
+        :param Context ctx: Context object.
+        :param Message request: Request message.
+        :param Message response: Response message.
+        :param str model_name: Name of the model in the Pipeline.models.
+        """
+        raise NotImplementedError
+
+
+class IsImportant(BaseFilter):
+    """
+    Filter that checks if the "important" field in a Message.misc is True.
+    """
+
+    def __call__(
+        self, ctx: Context = None, request: Message = None, response: Message = None, model_name: str = None
```

Comment:
https://github.com/deeppavlov/chatsky/pull/376/commits/9188b8972e7a60024efbee04bd91b0cd95dcaf9b

Reply:
I meant that they don't need a default value since filters are always called from our functions that generate history from context so we can make sure these arguments are always passed.

Removed defaults in 934a0b88adc7408fa820298bb845c2997ffa14a1.